si5351/si5351.go

```diff
@@ -0,0 +1,491 @@
+package si5351
+
+import (
+	"errors"
+	"fmt"
+	"math"
+
+	"tinygo.org/x/drivers"
+	"tinygo.org/x/drivers/internal/legacy"
```

Comment:
This is known to cause heap allocations, shall we try and avoid this from start?

Reply:
Updated again with the latest code from #801